requirements-dev.txt

```diff
@@ -5,6 +5,8 @@ git+https://github.com/asdf-format/asdf-unit-schemas.git
 git+https://github.com/asdf-format/asdf-wcs-schemas
 git+https://github.com/yaml/pyyaml.git
 
+pytest>=0.0.dev0
```

Comment:
should we run this in a different `dev` test?

Reply:
Thanks for giving this a review.

That sounds like a great idea! It does seem bad to have all our devdeps fail just because pytest makes some change. I am planning to update https://github.com/asdf-format/asdf/pull/1731 and bring it out of draft after this PR is merged which would install pytest from dev (further increasing the risk of a change breaking all the tests).

Do you have a suggestion for how to break this out? Maybe update tox with a `-pytestdev` option and add it to the CI? Thanks in advance for your help :)

Reply:
yeah I think a `-pytestdev` tox factor is a good idea, because then we could also theoretically mix and match it with `-devdeps` or `-stdevdeps`

Reply:
Good point! I'll update this PR.